Packs/AutoFocus/Integrations/AutoFocusTagsFeed/AutoFocusTagsFeed.py

```diff
@@ -0,0 +1,509 @@
+    return EntityRelationship(
+        name=MAP_RELATIONSHIPS.get(a_class, {}).get(b_class),
+        entity_a=a_name,
+        entity_a_type=a_class,
+        entity_b=b_name,
+        entity_b_type=b_class,
+        reverse_name=MAP_RELATIONSHIPS.get(b_class, {}).get(a_class),
+    )
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def test_module(client: Client) -> str:
+    """Builds the iterator to check that the feed is accessible.
+    Args:
+        client: Client object.
+    Returns:
+        Outputs.
+    """
+    try:
+        response = client.get_tags(data={'pageSize': 1})
+        tag_list = response.get('tags', [])
+        tag = tag_list[0] if len(tag_list) > 0 else {}
+        tag_name = tag.get('public_tag_name')
+        client.get_tag_details(tag_name)
+    except Exception as e:
+        raise DemistoException(str(e))
+    return 'ok'
+
+
+def fetch_indicators(client: Client,
+                     is_get_command: bool,
+                     tlp_color: Optional[str] = None,
+                     feed_tags: List = None,
+                     limit: int = -1) -> List[Dict]:
+    """Retrieves indicators from the feed
+    Args:
+        is_get_command: is this method called from the get-indicators-command
+        client (Client): Client object with request
+        tlp_color (str): Traffic Light Protocol color
+        feed_tags (list): tags to assign fetched indicators
+        limit (int): limit the results
+    Returns:
+        Indicators list.
+    """
+    iterator = client.build_iterator(is_get_command, limit)
+    indicators = []
+    for tag_details in iterator:
+        tag = tag_details.get('tag')
+        value_ = tag.get('tag_name')
+        tag_class = tag.get('tag_class')
+        source = tag.get('source')
+        type_ = get_tag_class(tag_class, source)
+        if not type_:
+            continue
+        raw_data = {
+            'value': value_,
+            'type': type_,
+        }
+        for key, value in tag_details.items():
+            raw_data.update({key: value})
+        indicator_obj = {
+            'value': value_,
+            'type': type_,
+            'service': 'AutoFocus',
+            'fields': get_fields(tag_details),
+            'rawJSON': raw_data,
+            'score': SCORES_MAP.get(type_)
+        }
+        related_tags = tag_details.get('related_tags', [])
+        if related_tags:
+            relationships = (create_relationships_for_tag(client, value_, type_, related_tags))
+            if relationships:
+                indicator_obj['relationships'] = relationships
+        tag_groups = get_tag_groups(tag_details.get('tag_groups', []))
+        if feed_tags or tag_groups:
+            if feed_tags:
+                tag_groups.extend(feed_tags)
+            indicator_obj['fields']['tags'] = tag_groups
+
+        if tlp_color:
+            indicator_obj['fields']['trafficlightprotocol'] = tlp_color
+        indicators.append(indicator_obj)
+    return indicators
```

Comment:
This method is built from a lot of simple logics - see if you can export them into small helper methods.
That should really improve the readability and simplicity of the code.